applications/Preprocessor/lib/mesh/MeshImpl/MeshEntity.h

```diff
@@ -173,7 +204,26 @@ class MeshEntity {
     /// \param elementID The entityID of the element
     /// \param localEntityIndex The localIndex of this MeshEntity for the
     /// element.
-    void addElement(EntityGId elementID, EntityLId EntityLId);
+    void addElement(EntityGId elementID, EntityLId localEntityIndex);
+
+    /// Reverse of addElement, remove an Element that this MeshEntity is part
+    /// of.
+    ///
+    /// \param elementId The id of the element to remove
+    /// \param localEntityIndex The local index on the element of this
+    /// MeshEntity (before removal).
```

Comment:
what happens at an id, after the element is removed?

Reply:
I've updated the documentation to make clear that this should only be used by `Element` to keep the link `MeshEntity -> Element` in sync with the link `Element -> MeshEntity` (and similar for addElement).

So it is the problem of the Element what happens at that specific index. The specific case where this is used is to overwrite the MeshEntity it links to. So it removes the link to one MeshEntity, calling this function to keep the reverse link up to date, ets a new MeshEntity and uses `MeshEntity#addElement` to set its link.

Reply:
maybe `Relink` is a better name?

Reply:
I see what you are saying, but it is not intuitive from the functionname